eventstore/views.py

```diff
@@ -20,6 +27,89 @@
+                        "audio": inbound.get("audio"),
+                        "document": inbound.get("document"),
+                        "image": inbound.get("image"),
+                        "system": inbound.get("system"),
+                        "location": inbound.get("location"),
+                        "text": inbound.get("text"),
+                        "video": inbound.get("video"),
+                        "voice": inbound.get("voice"),
+                    }
+                    Messages.objects.create(
+                        id=inbound["id"],
+                        contact_id=inbound["from"],
+                        type=inbound["type"],
+                        data=data,
+                        message_direction=Messages.INBOUND,
+                        created_by=request.user.username,
+                        recipient_type=inbound["recipient_type"],
+                        timestamp=datetime.fromtimestamp(int(inbound["timestamp"])),
+                    )
+
+                for statuses in request.data.get("statuses", []):
+                    data = {"errors": statuses.get("errors")}
+                    Events.objects.create(
+                        message_id=statuses.get("id"),
+                        recipient_id=statuses.get("recipient_id"),
+                        timestamp=datetime.fromtimestamp(
+                            int(statuses.get("timestamp"))
+                        ),
+                        status=statuses.get("status"),
+                        created_by=request.user.username,
+                        data=data,
+                    )
+
+            elif webhook_type == "turn":
+                inbound = request.data["messages"]
+                data = {
+                    "text": inbound.text,
+                    "render_mentions": inbound.render_mentions,
+                    "preview_url": inbound.preview_url,
+                }
+                Messages.objects.create(
+                    id=request.headers["X-WhatsApp-Id"],
+                    contact_id=inbound.to,
+                    type=type,
+                    data=data,
+                    message_direction=inbound.OUTBOUND,
+                    created_by=request.user.username,
+                    recipient_type=inbound.recipient_type,
+                )
+            else:
+                data = {
+                    "text": inbound.text,
+                    "render_mentions": inbound.render_mentions,
+                    "preview_url": inbound.preview_url,
+                }
+                Messages.objects.create(
+                    id=request.headers["X-WhatsApp-Id"],
+                    contact_id=inbound.to,
+                    type=type,
+                    data=data,
+                    message_direction=inbound.OUTBOUND,
+                    created_by=request.user.username,
+                    recipient_type=inbound.recipient_type,
+                )
```

Comment:
This seems to be just a copy of the previous code block.

I think instead of trying to parse the request, if the header doesn't contain a value that we recognise, we should return a 400 Bad Request error, explaining that we only accept `whatsapp` and `turn` webhook types.

Reply:
so, under the else, I could simple return a `return Response(status=status.HTTP_400_BAD_ERROR)`

Reply:
Yes, we'll probably also want to include a body explaining why we sent a 400